### PR TITLE
Fix parsing post data

### PR DESCRIPTION
**Pull request by girardinsamuel**
- [x] add failing test
- [x] fix behaviour

I have added a test for each content type including the one which was not working (i.e. `multipart/form-data`) so that we can better prevent regression errors.
It would maybe need more tests as it's dealing with user inputs and there are lot of edge cases, but it should be good for now.

### CODE REVIEW

File: src/masonite/input/InputBag.py [hunk 1]
```diff
@@ -56,11 +56,6 @@ def parse(self, environ):
                 self.post_data = self.parse_dict(parsed_request_body)
 
             elif "multipart/form-data" in environ.get("CONTENT_TYPE", ""):
-                try:
```
Comment (girardinsamuel) [inline src/masonite/input/InputBag.py:59]: not used

Comment (josephmancuso) [reply src/masonite/input/InputBag.py:59]: hmm. How do you know this isn't used? We may just not have a test for it.

I would assume in the event the browser does not pass a content length for the request this would throw an error right?

File: tests/core/request/test_input.py [hunk 2]
```diff
@@ -95,6 +126,21 @@ def test_can_parse_post_params_from_url_encoded(self):
...
 
+    def test_can_parse_multipart_formdata_content_type(self):
+        data, content_type = encode_multipart_formdata({"key": "value", "test": 1})
+        bag = InputBag()
+        bag.load(
+            {
+                "REQUEST_METHOD": "POST",
+                "CONTENT_TYPE": content_type,
+                "CONTENT_LENGTH": str(len(data.encode("latin-1"))),
+                "wsgi.input": io.BytesIO(data.encode("latin-1")),
```
Comment (josephmancuso) [inline tests/core/request/test_input.py]: why latin-1 here and not utf-8?

**Review by josephmancuso — CHANGES_REQUESTED** [review]

File: src/masonite/input/InputBag.py (hunk 1 above)
Comment (girardinsamuel) [reply src/masonite/input/InputBag.py:59]: I mean that the variable in the code is really not used later.
That's the difference with the other content types where we are using it.

File: tests/core/request/test_input.py (hunk 2 above)
Comment (girardinsamuel) [reply tests/core/request/test_input.py]: It's just that I used an example found on internet to simulate a multipart/form-data (took me hours to find this).
But maybe it will work with utf-8, I will check.

File: src/masonite/input/InputBag.py (hunk 1 above)
Comment (josephmancuso) [reply src/masonite/input/InputBag.py:59]: Oh ok cool lol

File: tests/core/request/test_input.py (hunk 2 above)
Comment (girardinsamuel) [reply tests/core/request/test_input.py]: that's fixed

**Review by josephmancuso — APPROVED** [review]
